AdminCore.FsmWorkflow/FsmMachines/WorkflowFsmWfh.cs

```diff
@@ -0,0 +1,122 @@
+using AdminCore.Common;
+using AdminCore.Constants.Enums;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveStates;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveTriggers;
+using AdminCore.FsmWorkflow.FsmMachines.FsmWorkflowState;
+using Stateless;
+
+namespace AdminCore.FsmWorkflow.FsmMachines
+{
+    public class WorkflowFsmWfh : WorkflowFsm<WfhState, LeaveTriggersWfh>
+    {       
+        private StateMachine<WfhState, LeaveTriggersWfh>.TriggerWithParameters<EventStatuses, string> LeaveResponseTrigger;
+        
+        public WorkflowFsmWfh(WorkflowStateData fsmStateData)
+        {           
+            ConfigureFsm(fsmStateData);
+        }
+        
+        public override void ConfigureFsm(WorkflowStateData fsmStateData)
+        {
+            FsmStateData = fsmStateData;
+            FsMachine = new StateMachine<WfhState, LeaveTriggersWfh>(() => (WfhState)FsmStateData.CurrentState, s => FsmStateData.CurrentState = (int)s);
+            
+            LeaveResponseTrigger = FsMachine.SetTriggerParameters<EventStatuses, string>(LeaveTriggersWfh.LeaveResponded);
+            
+            // Leave Awaiting Team Lead and Client
+            FsMachine.Configure(WfhState.LeaveAwaitingTeamLead)
+                .SubstateOf(WfhState.LeaveAwaitingResponses)
+                .OnActivate(() =>
+                {
+                    if (IsAdminResponseReceived(FsmStateData.ApprovalDict))
+                    {
+                        FsMachine.Fire(IsAdminResponseApprove(FsmStateData.ApprovalDict)
+                            ? LeaveTriggersWfh.AdminApprove
+                            : LeaveTriggersWfh.AdminReject);
+                    }
+                    else
+                    {
+                        switch (FsmStateData.ApprovalDict[FsmStateData.TeamLead])
```

Comment:
I think for readability sake storing properties like this with a meaningful `var` name would help

Reply:
Added a method FireApproveRejectBasedOnResponderResponse to increase readability here.